frontend/src/pages/CompareV2.tsx

```diff
@@ -530,9 +546,17 @@ function CompareV2(props: CompareV2Props) {
                     namespace={namespace}
                   />
                 )}
-                {/* TODO(zpChris): Add more ROC Curve selections through checkbox system. */}
                 {metricsTab === MetricsType.ROC_CURVE && (
-                  <ROCCurveMetrics selectedRocCurveArtifacts={selectedRocCurveArtifacts} />
+                  <RocCurveMetrics
+                    linkedArtifacts={rocCurveLinkedArtifacts}
+                    filter={{
+                      selectedIds: selectedRocCurveIds,
+                      setSelectedIds: setSelectedRocCurveIds,
+                      fullArtifactPathMap,
+                      selectedIdColorMap,
+                      setSelectedIdColorMap,
+                    }}
```

Comment:
These `filter` parameters are stored in the `CompareV2` file so the ROC Curve metrics tab retains its state even when switching to and returning from a different metrics tab on the same KFPv2 Run Comparison page.